### PR TITLE
refactor(DB): Handle SQL files in a new way

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
- This will not break the current CI which means we will still have automatic SQL files moved to the correct folder and renaming
- This will not break the script to generate SQL files
- This will not break DB Autoupdater
- This will 100% break DB Assambler (Which has been deprecated and not supported since a while back)
- This will 100% fix the issues we are having with larger SQL files which can be run when it's not a PROCEDURE
- This will remove a lot of clutter of already committed SQL files

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/11417

## Tests Performed:

- I have run create_sql.sh script in all the DB folders
- I have run ci-pending-sql.sh script which the CI uses to move the SQL file form the pendings folder to the updates folder and renaming it. This can be seen working for all databases here:
image

## How to Test the Changes:

1. Run create_sql.sh
2. Run ci-pending-sql.sh

## Known Issues and TODO List:

- [x] Before this gets merged we need to remove the files I have created in the /updates/ folder as proof that the CI will still work
- [x] We should definitely first squash the database before merging this PR so that all /updates/ folders are empty and so we start from scratch with clean base folders as well. It's about time for it anyways, contributions are flowing in =) <3

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Winfidonarleyan** [comment]
Not bad, need totaly test this

**Comment by Kitzunu** [comment]
@Winfidonarleyan please test it on your local <3

**Comment by Winfidonarleyan** [comment]
> @Winfidonarleyan please test it on your local <3

Remind me of that tomorrow.

**Comment by Kitzunu** [comment]
if I remember xD

**Comment by Kitzunu** [comment]
Should merge this after https://github.com/azerothcore/azerothcore-wotlk/pull/11515 so we have a clean updates folder to start from

**Review by Nyeriah — APPROVED** [review]

**Comment by Kitzunu** [comment]
> > @Winfidonarleyan please test it on your local <3
> 
> Remind me of that tomorrow.

**Comment by Nyeriah** [comment]
Hey guys,

No pressure but, we just had a squash, so it would be super cool if we could get the new SQL format in place before we're flooded with lots of SQL files using the old, broken procedure

**Comment by Winfidonarleyan** [comment]
@Kitzunu normal work, good job. Can merge

File: src/server/game/World/World.cpp [hunk 1]
```diff
@@ -3266,45 +3266,6 @@ void World::LoadDBVersion()
...
-{
-    QueryResult resultWorld     = WorldDatabase.Query("SELECT date FROM version_db_world ORDER BY date DESC LIMIT 1");
-    QueryResult resultCharacter = CharacterDatabase.Query("SELECT date FROM version_db_characters ORDER BY date DESC LIMIT 1");
-    QueryResult resultAuth      = LoginDatabase.Query("SELECT date FROM version_db_auth ORDER BY date DESC LIMIT 1");
-
-    if (resultWorld)
-    {
-        Field* fields = resultWorld->Fetch();
-
-        m_WorldDBRevision = fields[0].Get<std::string>();
```
Comment (Winfidonarleyan) [inline src/server/game/World/World.cpp:3279]: need delete this variables

File: src/server/game/World/IWorld.h [hunk 2]
```diff
@@ -582,7 +582,6 @@ class IWorld
     virtual void UpdateRealmCharCount(uint32 accid) = 0;
     [[nodiscard]] virtual LocaleConstant GetAvailableDbcLocale(LocaleConstant locale) const = 0;
     virtual void LoadDBVersion() = 0;
-    virtual void LoadDBRevision() = 0;
     [[nodiscard]] virtual char const* GetDBVersion() const = 0;
     [[nodiscard]] virtual char const* GetWorldDBRevision() const = 0;
     [[nodiscard]] virtual char const* GetCharacterDBRevision() const = 0;
```
Comment (Winfidonarleyan) [inline src/server/game/World/IWorld.h:587]: Need delete maybe?

**Comment by Yehonal** [comment]
Is the worldserver  the alternative to the db assembler?

I'm not into the new migration system anymore but ideally a database migration should be handled by a separated migration tool that should run separately, it allows you to spin up a database in many environments where the worldserver is not always available, moreover, it allows you to avoid having the full SQL files within the worldserver environment since it's not always possible.


Some examples:

* Developing a website on top of the AC database, I should not compile AC and run the entire worldserver to have a DB
* Deploying AC on a distributed docker environment -> it's expensive to push an image that contains gigabytes of SQL files that are only used when the database is updated. A best practice would be to have an "init container" that runs a tool to migrate the DB and it stops before the worldserver container starts.

Beside of using DB assembler or the C++ system, we should provide a small tool that is separated by the worldserver itself and that can be even integrated in a docker image together with the SQL files and a MySQL instance.

**Comment by Yehonal** [comment]
@Winfidonarleyan since you imported that system, it'd be possible for you to create a db_migrate executable that reuse the system integrated within the worldserver without all the worldserver sources?

It should be an executable of few megabytes that only runs the db migrations, in this way I can create a docker image that includes:

1. a MySQL instance with preloaded database
2. the new executable that runs right after the MySQL server
3. all the SQL updates

In this way we can have a docker image called `ac-database` that is able to update itself without the needs of having the worldserver executable

**Comment by Kitzunu** [comment]
🤦‍♂️ that is a good suggestion.

But we really need this now. That cn come in the future. And the Assambler is already not supported whoch has been decided by the majority pf the staff.

So i will look over Winfis review and then we merge this.

**Comment by Nyeriah** [comment]
I'll 

> 🤦‍♂️ that is a good suggestion.
> 
> But we really need this now. That cn come in the future. And the Assambler is already not supported whoch has been decided by the majority pf the staff.
> 
> So i will look over Winfis review and then we merge this.

I will second this.

The procedure breaks importing large database files and also disallows editing old files in case mistakes were made. It would be better if the solution to it was not conditioned by another tool that may or may not be scripted reasonably soon. That would inevitably and suddenly delay a solution that solves our problems, postponing its effect from soon to an indefinite date, which certainly isn't optimal but surely somewhat frustrating as well.

**Comment by Winfidonarleyan** [comment]
> 

I already wanted to do it, but then I forgot about it, I will try to do it in the next few days

**Comment by Winfidonarleyan** [comment]
> @Winfidonarleyan since you imported that system, it'd be possible for you to create a db_migrate executable that reuse the system integrated within the worldserver without all the worldserver sources?
> 
> It should be an executable of few megabytes that only runs the db migrations, in this way I can create a docker image that includes:
> 
> 1. a MySQL instance with preloaded database
> 2. the new executable that runs right after the MySQL server
> 3. all the SQL updates
> 
> In this way we can have a docker image called `ac-database` that is able to update itself without the needs of having the worldserver executable

I already wanted to do it, but then I forgot about it, I will try to do it in the next few days

**Comment by Winfidonarleyan** [comment]
> @Winfidonarleyan since you imported that system, it'd be possible for you to create a db_migrate executable that reuse the system integrated within the worldserver without all the worldserver sources?
> 
> It should be an executable of few megabytes that only runs the db migrations, in this way I can create a docker image that includes:
> 
> 1. a MySQL instance with preloaded database
> 2. the new executable that runs right after the MySQL server
> 3. all the SQL updates
> 
> In this way we can have a docker image called `ac-database` that is able to update itself without the needs of having the worldserver executable

I created this - https://github.com/WarheadCore/WarheadBand/commit/24981b4f6b15db6946adf1a14163272f4ca057b3

**Pull request merged**